### DISCUSSION TITLE
GitHub Pages URL parameters

### DISCUSSION THREAD

**Discussion by Galaxy-Coding** [community/community#22667] [Other Feature Feedback, Questions, & Ideas]
<p>I have an input in a HTML page(). The form should redirects to another URL with a url parameter that is equal to the input. The URL would look like this:<br>
<code>user.github.io/repository?color=red</code><br>
How could I do that?<br>
Thank you in advance!</p>

**Comment by SethClydesdale** [comment, marked as answer]
<p>Hello <a class="mention" href="/galaxy-coding">@Galaxy-Coding</a>, it depends on the form.</p>
<p>With HTML you can do:</p>
<pre><code class="lang-auto">&lt;form&gt;
  &lt;input type="text" name="color" value="red"&gt;&lt;br&gt;
  &lt;input type="submit" value="Submit"&gt;
&lt;/form&gt; 
</code></pre>
<p>If sending to a new page, you can add the <code>action</code> attribute to <code>&lt;form&gt;</code>. (e.g. action=“my_url”)</p>
<p>or with JavaScript:</p>
<pre><code class="lang-auto">&lt;form id="myForm"&gt;
  &lt;input type="text" name="color" value="red"&gt;&lt;br&gt;
  &lt;input type="submit" value="Submit"&gt;
&lt;/form&gt; 
&lt;script&gt;
document.getElementById('myForm').onsubmit = function () {
  // set params using form values
  window.location.search = this.color.name + '=' + this.color.value;
  
  // prevents normal form submission
  return false;
};
&lt;/script&gt;
</code></pre>
<p>If you’re setting the whole URL, you can use <code>location.href</code> instead and append your params.</p>
